src/data-workspace/inputs/generic-input.js

```diff
@@ -43,23 +43,47 @@ export const GenericInput = ({
         )
     }
 
+    const formatValue = (value) => {
+        if (value === undefined) {
+            return ''
+        } else if (
+            value.trim() &&
+            NUMBER_TYPES.includes(valueType) &&
+            Number.isFinite(Number(value))
+        ) {
+            return Number(value).toString()
+        } else {
+            return value.trim()
+        }
+    }
+
     const { input, meta } = useField(fieldname, {
         validate: validatorsByValueType[valueType],
         subscription: { value: true, dirty: true, valid: true },
+        format: formatValue,
+        formatOnBlur: true,
     })
 
+    // need to initialize lastSyncedValue
+    useEffect(() => {
+        if (input.value !== undefined) {
+            setLastSyncedValue(input.value)
```

Comment:
I found myself doing this to get the lastSyncedValue to behave nicely - if the initial value isn't set the user can't delete the the value from the form.